dynamic-simulation/dynamic-simulation-api/src/main/java/com/powsybl/dynamicsimulation/Provider.java

```diff
@@ -0,0 +1,37 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package com.powsybl.dynamicsimulation;
+
+import com.powsybl.iidm.network.Network;
+
+import java.util.List;
+
+/**
+ * @author Mathieu Bague <mathieu.bague@rte-france.com>
+ */
+public interface Provider<T> {
```

Comment:
What about naming this class as `DynamicSimulationDataProvider` or `DynamicSimulationInputsProvider` ?

Reply:
With @agnesLeroy we think this could be moved in a more common module, so we wouldn't add a Dynamic prefix. Looking to the Javadoc, the good name should be `Supplier`. The `Supplier` class is a generic class that have only one method `get`. In our case, it's the same concept, except that the `get` method takes one argument.